kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -154,4 +155,20 @@ public long getTimestamp(Engine engine) {
             return logSegment.lastCommitTimestamp;
         }
     }
+
+    /**
+     * Returns the commit coordinator client handler for this snapshot.
+     * If the snapshot is not configured to use a commit coordinator, this returns an empty
+     * Optional.
+     *
+     * @param engine the engine to use for IO operations
+     * @return the commit coordinator client handler for this snapshot
+     */
+    public Optional<CommitCoordinatorClientHandler> getCommitCoordinatorClientHandlerOpt(
```

Comment:
I wonder if it would better to replicate the `TableCommitCoordinator` from Delta-Spark in Kernel? `TableCommitCoordinator` wraps around CoordinatedCommitsHandler and stores logpath + configs so that you don't have to pass these to parameters to it on every commit/getCommits call. What do you think @vkorukanti ?